lasio/las.py

```diff
@@ -139,6 +141,11 @@ def add_section(pattern, name, **sect_kws):
                 logger.warning(
                     "Header section %s regexp=%s was not found." % (name, pattern)
                 )
+                if name == "Version":
+                    self.version_section_not_found = True
+                elif name == "Well":
+                    self.well_section_not_found = True
+
```

Comment:
Don't do the check just on the LAS file being loaded - the user might add sections later. The check should be done every time ``check_conforming()`` is called.

Reply:
The catch is, if a section doesn't exist the default value added in __init__ i.e. default_items["Version"], default_items["Well"] etc. persists. So should we delete the default value/s for sections in such case? Because otherwise the las file will always have version, well sections.

Reply:
Good point. I think the best solution is to only create the default sections if we are **not** reading a file in. e.g. something like cc0430. That commits breaks some tests, but I'm not worried. At the moment we are producing side-effects by reading a file in which is pretty odd.

Reply:
Ultimately I want to get to a place where `lasio.read("blah.las").write()` produces something identical, and `lasio.read("blah.las").to_conforming().write()` makes the necessary changes like added sections, cleaned up formatting, etc.

Reply:
> Good point. I think the best solution is to only create the default sections if we are **not** reading a file in. e.g. something like [cc0430b](https://github.com/kinverarity1/lasio/commit/cc0430bb1e59bee1459de2ac533508a641f48aed). That commits breaks some tests, but I'm not worried. At the moment we are producing side-effects by reading a file in which is pretty odd.

OK, I'll use this approach then. I'll fix those tests or remove if required.

Reply:
No worries. Thanks for working on this! 

I've looked at the failing tests. Don't worry about `test_write_wrapped` or `test_write_unwrapped` - I'll fix those later, they are poorly written. 

I'd like to have `test_barebones` and  `test_barebones_missing_all_sections` kept as they are, I can work on a fix for that too so don't worry about them continuing to fail.